docs/developerguide.rst

```diff
@@ -0,0 +1,53 @@
+.. _developerguide:
+
+Developer guide
+===============
+
+Deploying specific component versions
+-------------------------------------
+
+If you want to try different versions of REANA components, you can modify your
+``reana-cluster.yaml`` configuration file; see :ref:`configure`.
+
+You can check available released versions on `REANA DockerHub
+<https://hub.docker.com/u/reanahub/>`_.
+
+Deploying latest ``master`` branch versions
+-------------------------------------------
+
+If you want to use latest ``master`` branch of REANA components and have the
+code directly editable within pods, you can use ``reana-cluster-dev.yaml``.
+
+1. Clone the REANA component under ``$HOME/reana/sources`` directory on your laptop.
+
+2. Mount your component source code in minikube under `/code` directory:
+
+.. code-block:: console
+
+   $ ls -d $HOME/reana/sources
+   reana-client
+   reana-cluster
+   reana-job-controller
+   reana-message-broker
+   reana-server
+   reana-workflow-controller
+   reana-workflow-engine-cwl
+   reana-workflow-engine-yadage
+   reana-workflow-monitor
+   $ minikube mount $HOME/reana/sources:/code
+
+3. Edit sources on your laptop under ``$HOME/reana/sources/reana-<component>/...`` as usual:
+
+.. code-block:: console
+
+   $ vim $HOME/reana/sources/reana-job-controller/reana_job_controller/app.py
+
+4. Delete your pod and see a new pod being deployed with your changes:
+
+.. code-block:: console
+
+   $ kubectl get pods | grep job-controller
+   $ kubectl delete pod job-controller-2461563162-r4hgg
+   $ sleep 10 # wait some time...
+   $ kubectl get pods | grep job-controller
+   $ kubectl exec -i -t job-controller-2461563162-h9ptw /bin/bash
```

Comment:
@hjhsalo @tiborsimko  that means that the only thing we tested and advertise is that the code has been changed, did we also check if the application gets updated to the latest changes? (:

Reply:
Yes, that's the only tested thing so far. I've put it in the docs for illustration, so that @hjhsalo has a place to start adding docs. That said, we need to release 0.1.0 components and retest full chain with `reana-client` (=testing production-like conditions), and we need to merge @hjhsalo's devcomponents branch and retest say some simple API endpoint change and restart of the pod (=testing development-like conditions) to have the complete pictutre.

Reply:
Yes, correct. Only code changes but the code changes won't be picked up by the application since without building Dockerimages of REANA components without `"${DEBUG}" = "true"` the application is not installed in editable mode (https://github.com/reanahub/reana-server/blob/master/Dockerfile#L33).

I think we would change `reana-cluster-dev.yaml` to point to specific "development" images (e.g. `reana-workflow-controller:dev`) and provide instructions / convenience script / functionality in `reana-cluster` to build (and tag) the "development" images either 1) from latest master in Github or 2) from locally cloned repo (whatever branch is active).
These development images could be pushed to Dockerhub as well (e.g. build each time PR is merged to master).